sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/GpuHiveTextFileUtils.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.hive.rapids
+
+import org.apache.spark.sql.catalyst.expressions.AttributeReference
+import org.apache.spark.sql.types.{ArrayType, BinaryType, DataType, MapType, StructField, StructType}
+
+object GpuHiveTextFileUtils {
+  val textInputFormat      = "org.apache.hadoop.mapred.TextInputFormat"
+  val textOutputFormat     = "org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat"
+  val lazySimpleSerDe      = "org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe"
+  val serializationKey     = "serialization.format"
+  val ctrlASeparatedFormat = "1" // Implying '^A' field delimiter.
+  val lineDelimiterKey     = "line.delim"
+  val escapeDelimiterKey   = "escape.delim"
+  val newLine              = "\n"
+
+  def isSupportedType(dataType: DataType): Boolean = dataType match {
+    case ArrayType(_,_) => false
+    case StructType(_)  => false
+    case MapType(_,_,_) => false
+    case BinaryType     => false
+    case _              => true
```

Comment:
nit: This is protected further by the type checks when we translate it to the GPU. This is used in very different ways from that code, so it might be nice to document here and there that if you update one you might need to update the other too.

Reply:
Added the documentation on both ends, to keep them in sync.